multiversx/mx-sdk-nestjs-cli
# add start apps mxnest command

## Type


## Problem setting
- Currently, to start multiple NestJS applications from a monorepo, users need to manually open separate terminals and run the npm run start:<app> command for each application.

## Proposed Changes
- Implement a CLI command that uses `tmux` to start multiple NestJS applications from a monorepo in separate terminal sessions.
- Add a new start command to `nestjs-cli` that accepts a list of applications and starts each application in a separate `tmux` window.
-  Add a new stop command to `nestjs-cli` to stop all the nestjs-apps and also tmux session 
- Check for the presence of `tmux` on the system before starting the applications and display an error message if `tmux` `is not installed`.

## How to test
- Ensure `tmux` is installed on your system: Linux: `sudo apt-get install tmux` || macOs: `brew install tmux`
- Add the following script to your project's package.json to start all apps: `"apps:start": "mxnest start"`
- Add the following script to your project's package.json: to stop apps ( also opened tmux session ) `"apps:stop": "mxnest stop"`
- After you start the apps to see the apps terminals run command : `tmux attach-session -t nestjs-apps`
